src/restrictingGuards.ts

```diff
@@ -0,0 +1,97 @@
+import {checkerToGuard} from './Checker';
+import {thenGuard, andGuard, orGuard} from './combinators';
+import {ReasonGuard} from './ReasonGuard';
+import {isNumber, isSymbol, isString} from './primitiveGuards';
+
+export const numberIsInteger = checkerToGuard<number, number>((input: number) => {
+	if (!Number.isInteger(input)) {
+		throw new Error(`${input} is not an integer`);
+	}
+	return `${input} is an integer`;
+});
+
+export const numberIsLessThan = (maximum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input >= maximum) {
```

Comment:
I feel like it would be easier to check these guards if the conditions matched the names -- i.e. make the conditional target the positive case instead of an inverted condition for the negative case

Reply:
Traditionally all the code in this repo has been check-error ? throw : return success, e.g. `if ( typeof x !== 'foo')`. So I decided to be consistent here.